dipy/reconst/csdeconv.py

```diff
@@ -118,18 +120,20 @@ def __init__(self, gtab, response, reg_sphere=None, sh_order=8, lambda_=1,
         # scale lambda_ to account for differences in the number of
         # SH coefficients and number of mapped directions
         # This is exactly what is done in [4]_
-        self.lambda_ = (lambda_  * self.R.shape[0] * r_rh[0] /
-                        (np.sqrt(self.B_reg.shape[0]) * np.sqrt(362.))
-                       )
+
+        lambda_ = (lambda_  * self.R.shape[0] * r_rh[0] /
+                   (np.sqrt(self.B_reg.shape[0]) * np.sqrt(362.)))
```

Comment:
On Tue, Jan 27, 2015 at 2:00 PM, Samuel St-Jean <notifications@github.com>
wrote:

> In dipy/reconst/csdeconv.py
> <https://github.com/nipy/dipy/pull/553#discussion_r23647036>:
>
> > @@ -118,18 +120,20 @@ def __init__(self, gtab, response, reg_sphere=None, sh_order=8, lambda_=1,
> >          # scale lambda_ to account for differences in the number of
> >          # SH coefficients and number of mapped directions
> >          # This is exactly what is done in [4]_
> > -        self.lambda_ = (lambda_  * self.R.shape[0] * r_rh[0] /
> > -                        (np.sqrt(self.B_reg.shape[0]) * np.sqrt(362.))
> > -                       )
> > +
> > +        lambda_ = (lambda_  * self.R.shape[0] * r_rh[0] /
> > +                   (np.sqrt(self.B_reg.shape[0]) * np.sqrt(362.)))
>
> Maybe I got this all wrong, but what I get from this is :
>
>    1. We need a sqrt(362) to accomodate the old hardcoded regularisation
>    sphere.
>
> The sphere was never hard-coded, we just didn't take the number of points
on the sphere into account when we computed lambda. We do now.


>
>    1. Instead of changing that, users can play with $lambda'$, which
>    kinda takes into account that hardcoded value.
>
> Users are free to play with lambda', it's an agrument to the model.

>
>    1. We could change sqrt(362) to actually be the number of point on the
>    regularisation sphere, also defaulting to 362 for the default code path,
>    but also taking into account any other sampled sphere. Granted, anyone
>    using another sphere will get a deffiretn value from now on, but that's
>    correcting a hardcoded value at the same time, which is more sane in the
>    long run.
>
> Thanks what we do, if you look at the code it's
`np.sqrt(self.B_reg.shape[0]) * np.sqrt(362.)`, $N_0$ is just a constant,
it's the first part that's important. We can set $N_0$ to anything. I think
you're suggesting setting it to 1., but I 1. is no better than 20. is no
better than sqrt(362), it's just a constant.


>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nipy/dipy/pull/553/files#r23647036>.
>